includes/class-sanitize.php

```diff
@@ -201,4 +201,62 @@ public static function content( $content ) {
 
 		return $content;
 	}
+
+	/**
+	 * Remove hashtags from content.
+	 *
+	 * Removes hashtags that appear at the end of the content.
+	 * Handles both plain text and HTML content, including hashtags within anchor tags.
+	 *
+	 * @param string $content  The content to process.
+	 * @param array  $hashtags Array of hashtag strings (with or without #).
+	 *
+	 * @return string The content with trailing hashtags removed.
+	 */
+	public static function remove_hashtags( $content, $hashtags ) {
+		if ( empty( $content ) || empty( $hashtags ) || ! \is_array( $hashtags ) ) {
+			return $content;
+		}
+
+		// Normalize hashtags: remove # prefix and trim whitespace.
+		$normalized_tags = \array_map(
+			function ( $tag ) {
+				return \trim( \ltrim( $tag, '#' ) );
+			},
+			$hashtags
+		);
+
+		// Remove empty tags.
+		$normalized_tags = \array_filter( $normalized_tags );
+
+		if ( empty( $normalized_tags ) ) {
+			return $content;
+		}
+
```

Comment:
Let's make sure `extract_hashtags_from_activity_tags()` gives us content we don't have to normalize again.